File: typescript/src/address/format/formatResult.ts
import { RestApiAddressFormatResponse } from "../../server/address/format/restApiAddressFormatResponse";
import { AddressConfidence, lookupConfidence } from "../addressConfidence";
import { AddressComponents, restApiResponseToAddressComponents } from "./addressComponents";
import { AddressEnrichment, restApiResponseToAddressEncrichment } from "./addressEnrichment";
import { AddressFormatted, restApiToAddressFormatted } from "./addressFormatted";
import { AddressMetadata, restApiResponseToAddressMetadata } from "./addressMetadata";
import { FormatAddress, restApiResponseToFormatAddress } from "./formatAddress";

export type FormatResult = {
    globalAddressKey?: string;
    confidence?: AddressConfidence;
    address?: FormatAddress;
    addressFormatted?: AddressFormatted;
    components?: AddressComponents;
    metadata?: AddressMetadata;
    enrichment?: AddressEnrichment;
};

export function restApiResponseToFormatResult(response: RestApiAddressFormatResponse): FormatResult {
    
    const result: FormatResult = {};
    
    const apiResponse = response.result;
    const apiMetadata = response.metadata;
    const apiEnrichment = response.enrichment;

    if (apiResponse) {
        result.globalAddressKey = apiResponse.global_address_key;
        result.confidence = lookupConfidence(apiResponse.confidence);
        if (apiResponse.address) {
            result.address = restApiResponseToFormatAddress(apiResponse.address);
        }
        if (apiResponse.addresses_formatted && apiResponse.addresses_formatted.length > 0) {
            result.addressFormatted = restApiToAddressFormatted(apiResponse.addresses_formatted[0]);
        }
        if (apiResponse.components) {
            result.components = restApiResponseToAddressComponents(apiResponse.components);
        }       
    }
    if (apiMetadata) {
        result.metadata = restApiResponseToAddressMetadata(apiMetadata);
    }
    if (apiEnrichment) {
        result.enrichment = restApiResponseToAddressEncrichment(apiEnrichment);
    }

    return result;

}

File: csharp/DVSClient/DVSClient/address/format/FormatResult.cs
using DVSClient.Common;
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format
{
    public class FormatResult
    {
        public string? GlobalAddressKey { get; }
        public AddressConfidence? Confidence { get; }
        public FormatAddress? Address { get; }
        public AddressFormatted? AddressFormatted { get; }
        public AddressComponents? Components { get; }
        public AddressMetadata? Metadata { get; }
        public AddressEnrichment? Enrichment { get; }

        public FormatResult()
        {
        }

        public FormatResult(RestApiAddressFormatResponse response)
        {
            var apiResult = response.Result;
            var apiMetadata = response.Metadata;
            var apiEnrichment = response.Enrichment;

            if (apiResult != null)
            {
                GlobalAddressKey = apiResult.GlobalAddressKey;
                Confidence = apiResult.Confidence?.GetEnumValueFromJsonName<AddressConfidence>();
                Address = apiResult.Address != null ? new FormatAddress(apiResult.Address) : null;
                AddressFormatted = apiResult.AddressesFormatted != null ? new AddressFormatted(apiResult.AddressesFormatted.First()) : null;
                Components = apiResult.Components != null ? new AddressComponents(apiResult.Components) : null;
            }
            else
            {
                GlobalAddressKey = string.Empty;
                Confidence = null;
                Address = null;
                AddressFormatted = null;
                Components = null;
            }

            Metadata = apiMetadata != null ? new AddressMetadata(apiMetadata) : null;
            Enrichment = apiEnrichment != null ? new AddressEnrichment(apiEnrichment) : null;
        }
    }
}